src/odemis/driver/xt_client.py

```diff
@@ -2279,6 +2286,58 @@ def _moveTo(self, future: futures.Future, pos: Dict[str, float], rel: bool = Fal
                 # know the user that the move is not complete.
                 if future._must_stop.is_set():
                     raise CancelledError()
+
+                # TODO: extend the behavior if rotation, tilt axes also need to be checked.
+                # So far checking the above is not required, as we have observed reporting stage position only in x/y
+                # being compromised. The issue with case western system was that the stage metadata was not
+                # updated in timely manner during the overview acquisition in FM mode (during which rotation, tilt are
+                # not changed), resulting in misplaced tiles in the image display. Even when the stage has physically
+                # moved to the target position but took time in updating its actual position through metadata
+                # The issue is resolved by checking if only x and y stage metadata reports the target position properly
+
+                # The stage is not moving anymore, but we still want to wait until the position has been updated
+                # TODO: base the timeout on the stage movement time estimation (est. *10)
+                timeout = 10  # s
+                expected_end_time = time.time() + timeout  # s
+                check_pos = True
+                wait_duration = 20e-03  # s
```

Comment:
Can you move this up, and use it also in the `sleep`s of lines 2268 & 2277.
It be also clearer to use uppercases, to that it's clear it's a constant.